### ISSUE TITLE
VSCode FileDecoration not supported in custom tree view

### ISSUE THREAD

**Issue by SanjulaGanepola** [eclipse-theia/theia#9796]
### Bug Description:
When implementing a custom TreeView in a VSCode extension and rendering FileDecorations on the tree items, VSCode will show the decorations. When running in Theia, it is not being shown.

### Steps to Reproduce:
1. Build and run the [Theia browser example](https://github.com/theia-ide/theia/blob/master/doc/Developing.md#quick-start)
2. Into your workspace, clone https://github.com/SanjulaGanepola/vscode-extension-samples
3. Open `vscode-extension-samples/tree-view-sample/`
4. Run `npm install`
5. Run the extension using `F1` > `Hosted Plugin: Start Instance`
6. Open the Explorer and bring up the Test View.
7. Observe the tree items under the Test View are not highlighted in red and do not have the “?” badge.

Theia:
![image](https://user-images.githubusercontent.com/32170854/127376938-d20b9570-089e-4026-b1ef-067de4ece02b.png)

VSCode:
![image](https://user-images.githubusercontent.com/32170854/127376723-408a0c64-914f-40aa-bb36-0c3d4767cd0f.png)

**Comment by vince-fugnitto** [comment]
@SanjulaGanepola thank you for the discussion, please see if https://github.com/eclipse-theia/theia/pull/9643 resolves your issue.

**Comment by vince-fugnitto** [comment]
The issue is a duplicate of https://github.com/eclipse-theia/theia/issues/9642.

**Comment by SanjulaGanepola** [comment]
> @SanjulaGanepola thank you for the discussion, please see if #9643 resolves your issue.

I do not think this pull request can solve my issue because my goal is to use the file decorator rather than the problem decorator.

In terms of #9642, is there any progress on this issue?
